### PR TITLE
Add demux algorithm from JSEP

### PR DESCRIPTION
**Pull request by pthatcherg**

### CODE REVIEW

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 1]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
           from SSRC to identification-tag is received. Note that the mapping
           from SSRC to identification-tag can change at any time during an RTP
           session.
+	</t>
+
+
+      <t>Note: The following algorithm does not yet have WG consensus
+      but is included here as something concrete for the working group
+      to discuss.</t>
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Please remove this statement. It does not belong even in proposals for text in drafts in post-WG-last-call state. (this and some of the other comments in this review come from https://github.com/rtcweb-wg/jsep/issues/431)

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 2]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
...
+
+          <t>Construct a table mapping outgoing SSRC to "m=line" for
+          each "m=" line in this BUNDLE group and for each SSRC
+          configured for sending in that "m=" line.</t>
+
+          <t>Construct a table mapping payload type to "m=" line for
+          each "m=" line in the BUNDLE group and for each payload type
+          configured for receiving in that "m=" line.  If any payload
+          type is configured for receiving in more than one "m=" line
+          in the BUNDLE group, do not it include it in the table.</t>
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: suggested reformulation: "do not include it in the table" -> "mark it as "not usable" in the table".

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 3]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
...
+
+      <t>To prepare for demultiplexing RTP packets to the correct "m="
+      line, the following steps MUST be followed for each BUNDLE
+      group.</t>
+
+      <t>
+        <list>
+
+          <t>Construct a table mapping MID to "m=" line for each
+          "m=" line in this BUNDLE group.</t>
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Add comment: "Each MID maps to only one m= line, and each m= line maps to only one MID".

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 4]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
...
+
+          <t>Construct a table mapping incoming SSRC to "m=" line for
+          each "m=" line in this BUNDLE group and for each SSRC
+          configured for receiving in that "m=" line.</t>
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Add comment: "More than one SSRC can map to each m= line".

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 5]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
...
+          mapping MID to "m=" line, update the incoming SSRC mapping
+          table to include an entry that maps the packet's SSRC to the
+          "m=" line for that MID.</t>
+
+          <t>If the packet's SSRC is in the incoming SSRC mapping
+          table, route the packet to the associated "m=" line and
+          stop.</t>
+
+          <t>If the packet's payload type is in the payload type
+          table, update the the incoming SSRC mapping table to include
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: "... table, update ..." => "table, and marked as usable, update ..." (based on "usability" concern above)

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 6]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
...
+      MUST be performed to route the packet:</t>
+
+      <t>
+        <list>
+          <t>If the packet is of type SR, and the sender SSRC for the
+          packet is found in the incoming SSRC table, deliver a copy of the
+          packet to the "m=" line associated with that SSRC. In addition,
+          for each report block in the report whose SSRC is found in the
+          outgoing SSRC table, deliver a copy of the RTCP packet to the
+          "m=" line associated with that SSRC.</t>
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Linguistic: an m= line is not an active object. Suggest "to the m= line" => "to the handler designated for the m= line".

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml [hunk 7]
```diff
@@ -1119,21 +1114,115 @@ SDP Answer
...
+          "m=" line for that payload type. In addition, route the packet
+          to the associated "m=" line and stop.</t>
+
+          <t>Otherwise, drop the packet.</t>
+        </list>
+      </t>
+
+      <t>For each RTCP packet received (including each RTCP packet that is part
+      of a compound RTCP packet), the following type-specific handling
+      MUST be performed to route the packet:</t>
```
Comment (alvestrand) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: As I said in Seoul, I don't think claiming to have an exhaustive list of the RTCP handlers needed in this spec is a Good Thing. Suggested reformulation:

", the packet MUST be routed to the appropriate handler for the SSRCs it contains information about. Some examples of such handling are given below."

This makes it clear to implementors that when they encounter (for example) TMMBR packets, they have to route them reasonably, even though TMMBR is not listed below.

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 1 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Done

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 3 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: There is no m= line => MID mapping, but I could make it doubly clear that an m= line has only one MID.  

Further *all* the tables have only one mapping for each key.  I can just make that clear with a preceding paragraph.

...

OK, I just added two sentences to make this all doubly clear.

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 4 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Done

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 6 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: I made it even more clear than that, based on input from Cullen.  Take a look.

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 2 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: What does marking something as unusable in the table mean?  Put some special "nothing here" marker?  And what effect does that have on the processing different than leaving it out?

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 5 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: I'm not sure what "marked as usable" means other than having a value that's not empty/unusable.

File: draft-ietf-mmusic-sdp-bundle-negotiation.xml (hunk 7 above)
Comment (pthatcherg) [inline draft-ietf-mmusic-sdp-bundle-negotiation.xml]: Done.

**Pull request closed without merging**